pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -162,14 +158,50 @@ func (f *ring1Factory) Describer(mapping *meta.RESTMapping) (kubectl.Describer,
 			return &kubectl.ClusterDescriber{Interface: fedClientSet}, nil
 		}
 	}
+
+	// make a generic describer if we can, or return the original error
+	makeGenericDescriberOrErr := func(originalErr error) (kubectl.Describer, error) {
```

Comment:
Why inline this function?  Is it just for the closure scoping?  Generally I prefer breaking out pieces of logic into separate functions when possible.